### ISSUE TITLE
Handle rejecting transactions and add loading state buttons to UI 

### ISSUE THREAD

**Issue by arrenv** [JoinColony/colonyCDapp#2389]
Teamwork task: [Handle rejecting transactions and add loading state buttons to UI](https://tw.auroracreation.com/#/tasks/15914354)

## Steps to reproduce

1. Ensure you test while connected to Metamask.
2. Create a Payment builder action.
3. Get to the Fund step and click the fund button.
4. You will be able to click the button twice before signing.
5. Reject the transaction, you will be stuck in the loading state and won't be able to fund the payment.

This is in relation to this issue https://github.com/JoinColony/colonyCDapp/issues/2372

## Expected behaviour

* The button should still be visible just in a loading state, like other buttons that have a loading state.

![image](https://github.com/JoinColony/colonyCDapp/assets/33682027/a055bf09-0e14-4dd4-bc7b-5bfbccfffe3c)

![image](https://github.com/JoinColony/colonyCDapp/assets/33682027/9e878ef3-49be-480d-a51c-5db490f64d56)

You can see the design here to show it in it's loading state - [Figma link](https://www.figma.com/design/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?node-id=24247-197899&t=7I7TjYqeYeBBOsxp-4)

## Actual behaviour

* The loading state does not align with other loading state and disabled buttons.
* If you reject the transaction or it fails, the loading state remains.

![image](https://github.com/JoinColony/colonyCDapp/assets/33682027/99f8a38e-38a6-4648-94d3-0594a912c8aa)

**Comment by AdrianDudko** [comment]
@arrenv 

I'm putting this "On hold" because there's an issue with the sagas, of which @jakubcolony  is aware.
This will be picked up again by @joanna-pagepro or @CzarekDryl  once unblocked by @jakubcolony
